ui/lib/config-ui/addon/components/messages/page/create-and-edit-message-form.hbs

```diff
@@ -0,0 +1,174 @@
+            />
+            <Input
+              @type="text"
+              @value={{@message.linkHref}}
+              placeholder="Paste URL (e.g. www.learnmore.com)"
+              id="link-href"
+              class="input"
+              data-test-field="linkHref"
+              {{on "input" (pipe (pick "target.value") (fn (mut @message.linkHref)))}}
+            />
+          </div>
+        </div>
+      {{else if (eq attr.name "startTime")}}
+        <label for="message-start" class="has-text-weight-bold is-size-8">
+          Message starts
+        </label>
+        <Hds::Text::Body @tag="p" @size="200" class="has-bottom-margin-s">
+          Defaults to 12:00 a.m. the following day (local timezone).
+        </Hds::Text::Body>
+        <Input
+          @type="datetime-local"
+          @value={{@message.startTime}}
+          class="input has-top-margin-xs is-auto-width is-block"
+          name="startTime"
+          data-test-input="startTime"
+          {{on "input" this.updateDateTime}}
+        />
+      {{else if (eq attr.name "endTime")}}
+        <Hds::Form::Radio::Group @layout="vertical" @name="endTime" class="has-top-margin-l" as |G|>
+          <G.Legend>Message expires</G.Legend>
+          <G.Radio::Field @id="never" checked @value="" {{on "change" (fn (mut @message.endTime) "")}} as |F|>
+            <F.Label>Never</F.Label>
+            <F.HelperText>This message will never expire unless manually deleted by the operator.</F.HelperText>
+          </G.Radio::Field>
+          <G.Radio::Field @id="specific-date" @value="" {{on "change" (fn (mut @message.endTime) @message.endTime)}} as |F|>
+            <F.Label>Specific date</F.Label>
+            <F.HelperText>
+              This message will expire at midnight (local timezone) at the specific date.
+              <div class="control">
+                <Input
+                  @type="datetime-local"
+                  @value={{@message.endTime}}
+                  class="input has-top-margin-xs is-auto-width"
+                  name="endTime"
+                  data-test-input="endTime"
+                  {{on "input" this.updateDateTime}}
+                />
+              </div>
+            </F.HelperText>
+          </G.Radio::Field>
+        </Hds::Form::Radio::Group>
+      {{else}}
+        <FormField
+          @attr={{attr}}
+          @model={{@message}}
+          @modelValidations={{this.modelValidations}}
+          class="has-bottom-margin-m"
+          data-test-field={{true}}
+        />
+      {{/if}}
+    {{/each}}
+
+    <Hds::ButtonSet class="has-top-margin-s has-bottom-margin-m has-top-margin-xl">
+      {{! TODO: VAULT-21533 preview modal }}
+      <Hds::Button @text="Preview" @color="tertiary" @icon="eye" />
+
+      <Hds::Button
+        @text="{{if @message.isNew 'Create' 'Edit'}} message"
+        disabled={{(or (not @message.title) (not @message.message) (this.save.isRunning))}}
```

Comment:
In case more required fields are added you could use the `isValid` property form the model validations rather than checking if title and message have been set. It looks like you are showing `invalidFormMessage` in that case so you could do `{{or this.invalidFormMessage this.save.isRunning}}`.